docs/tutorials/deadlock.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env
+
+import os
+import threading
+import time
+
+
+def process1(lock1, lock2):
+    print("Process 1: Trying to acquire lock1")
+    lock1.acquire()
```

Comment:
Can you please use the context manager instead of the `acquire` method? Something like:

```
print("Thread 1: Trying to acquire lock1")
with lock1:
    print("Thread 1: lock1 acquired")

    time.sleep(1)

    print("Thread 1: Trying to acquire lock2")
    with lock2:
        print("Thread 1: lock2 acquired")
print("Thread 1: Released both locks")
```

I think this makes the code a bit more explicit and probably removes the need for comments.